src/util/autofilereloader.cpp

```diff
@@ -0,0 +1,23 @@
+#include "util/autofilereloader.h"
+
+#include <QDebug>
+#include <QString>
+
+#include "moc_autofilereloader.cpp"
+#include "util/runtimeloggingcategory.h"
+
+AutoFileReloader::AutoFileReloader(const RuntimeLoggingCategory& loggingCategory)
+        : m_loggingCategory(loggingCategory) {
+    connect(&m_fileWatcher,
+            &QFileSystemWatcher::fileChanged,
+            this,
+            &AutoFileReloader::slotFileChanged);
+}
+
+void AutoFileReloader::slotFileChanged(const QString& changedFile) {
+    qCDebug(m_loggingCategory) << "File" << changedFile << "has been changed.";
+    if (m_pathTriggeringReload.remove(changedFile)) {
+        m_fileWatcher.removePath(changedFile);
```

Comment:
removePath returns also true on success. What is the role for m_pathTriggeringReload?

Reply:
to ensure we don't emit a second signal in case of the append/truncate process (see the doc in the header).

Reply:
@daschuer this is to specifically fix #13083.
When I implemented this auto reload, I wanted to make sure not to introduce the same limitation on QML.

Reply:
iIRC, my initial implementation was exclusively relying on the file watcher to keep track of which files are being watched, but I can remember why I ended adding that extra array. I believe the initial PR/commits may have the explanation, I will try to dig it later, when I'm not AFK.

Reply:
So the [original implementation](https://github.com/mixxxdj/mixxx/pull/12795/files#diff-f9cb0d49677f8a2d66f91e60c0d4c802bc51bdb25161a65118bbdf884a8d3d7c) only contained the QFileSystemWatcher, but we [had to add](https://github.com/mixxxdj/mixxx/pull/12844/files#diff-f9cb0d49677f8a2d66f91e60c0d4c802bc51bdb25161a65118bbdf884a8d3d7c) the array to desperately handle acknowledgement as [QFileSystemWatcher::removePath](https://doc.qt.io/qt-6/qfilesystemwatcher.html#removePath)'s returns appears to not be consistent against platforms.